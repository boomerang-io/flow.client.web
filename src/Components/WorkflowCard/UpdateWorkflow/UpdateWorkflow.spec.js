import React from "react";
import ImportWorkflow from ".";
import { fireEvent } from "@testing-library/react";

const mockfn = jest.fn();

const props = {
  fetchTeams: mockfn,
  handleImportWorkflow: mockfn,
  workflowId: "test",
};

beforeEach(() => {
  document.body.setAttribute("id", "app");
});

describe("ImportWorkflow --- Snapshot Test", () => {
  it("Capturing Snapshot of ImportWorkflow", () => {
    const { baseElement, getByText } = rtlContextRouterRender(<ImportWorkflow {...props} />);
    fireEvent.click(getByText(/Choose a file or drag one here/i));
    expect(baseElement).toMatchSnapshot();
  });
});
