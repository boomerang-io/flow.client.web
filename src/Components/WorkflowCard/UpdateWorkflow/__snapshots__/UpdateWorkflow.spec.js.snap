// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportWorkflow --- Snapshot Test Capturing Snapshot of ImportWorkflow 1`] = `
<body
  class="bx--bmrg-body-modal-is-open"
  id="app"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="bx--bmrg-modal-portal"
  >
    <div
      class="bx--bmrg-modal-overlay --is-open"
    >
      <div
        aria-modal="true"
        class="bx--bmrg-modal-container bx--bmrg-modal-flow-container container --is-open"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="bx--modal-header"
        >
          <p
            class="bx--modal-header__heading bx--type-beta"
          >
            Update Workflow
          </p>
          <button
            aria-label="Close"
            class="bx--modal-close"
            title="Close"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="bx--modal-close__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
              />
            </svg>
          </button>
        </div>
        <form
          class="bx--bmrg-modal-flow-form"
        >
          <p
            class="bx--bmrg-modal-flow-form__title"
          >
            Select the Workflow JSON file you want to update the current Workflow with. The Workflow id must match.
          </p>
          <div
            class="bx--modal-content"
            style="height: 21rem; width: 100%;"
          >
            <div
              class="bx--file"
            >
              <label
                class="bx--file-browse-btn"
                for="id1"
                tabindex="0"
              >
                <div
                  class="bx--file__drop-container"
                  role="button"
                >
                  Choose a file or drag one here
                  <input
                    accept=".json"
                    class="bx--file-input"
                    id="id1"
                    name="Workflow"
                    tabindex="-1"
                    type="file"
                  />
                </div>
              </label>
            </div>
          </div>
          <div
            class="bx--modal-footer bx--btn-set"
          >
            <button
              class="bx--btn bx--btn--secondary"
              tabindex="0"
              type="button"
            >
              Cancel
            </button>
            <button
              class="bx--btn bx--btn--primary bx--btn--disabled"
              disabled=""
              tabindex="0"
              type="submit"
            >
              Update
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    class="bx--bmrg-modal-portal"
  />
</body>
`;
